analyzer/windows/modules/packages/xls.py

```diff
@@ -3,67 +3,24 @@
 # This file is part of Cuckoo Sandbox - http://www.cuckoosandbox.org
 # See the file 'docs/LICENSE' for copying permission.
 
-from _winreg import HKEY_CURRENT_USER
-
 from lib.common.abstracts import Package
+from lib.common.rand import random_string
 
-class XLS(Package):
-    """Excel analysis package."""
-    PATHS = [
-        ("ProgramFiles", "Microsoft Office", "EXCEL.EXE"),
-        ("ProgramFiles", "Microsoft Office", "Office10", "EXCEL.EXE"),
-        ("ProgramFiles", "Microsoft Office", "Office11", "EXCEL.EXE"),
-        ("ProgramFiles", "Microsoft Office", "Office12", "EXCEL.EXE"),
-        ("ProgramFiles", "Microsoft Office", "Office14", "EXCEL.EXE"),
-        ("ProgramFiles", "Microsoft Office", "Office15", "EXCEL.EXE"),
-        ("ProgramFiles", "Microsoft Office", "Office16", "EXCEL.EXE"),
-        ("ProgramFiles", "Microsoft Office 15", "root", "office15", "EXCEL.EXE"),
-        ("ProgramFiles", "Microsoft Office", "root", "Office16", "EXCEL.EXE"),
-    ]
-
-    REGKEYS = [
-        [
-            HKEY_CURRENT_USER,
-            "Software\\Microsoft\\Office\\14.0\\Common\\General",
-            {
-                # "Welcome to the 2010 Microsoft Office system"
-                "ShownOptIn": 1,
-            },
-        ],
-        [
-            HKEY_CURRENT_USER,
-            "Software\\Microsoft\\Office\\14.0\\Excel\\Security",
-            {
-                # Enable VBA macros in Office 2010.
-                "VBAWarnings": 1,
-                "AccessVBOM": 1,
 
-                # "The file you are trying to open .xyz is in a different
-                # format than specified by the file extension. Verify the file
-                # is not corrupted and is from trusted source before opening
-                # the file. Do you want to open the file now?"
-                "ExtensionHardening": 0,
+class XLS(Package):
+    """Excel analysis package.
+    We hack excel single instance by loading sample from commandline
+    """
 
-                # "Data connection has been blocked"
-                "DataConnectionWarnings": 0,
-                "WorkbookLinkWarnings": 0,
-            },
-        ],
-        [
-            HKEY_CURRENT_USER,
-            "Software\\Microsoft\\Office\\Common\\Security",
-            {
-                # Enable all ActiveX controls without restrictions & prompting.
-                "DisableAllActiveX": 0,
-                "UFIControls": 1,
-            },
-        ],
+    PATHS = [
+        ("System32", "cmd.exe"),
     ]
 
     def start(self, path):
         self._allow_embedded_flash()
-
-        excel = self.get_path("Microsoft Office Excel")
-        return self.execute(
-            excel, args=[path], mode="office", trigger="file:%s" % path
-        )
+        # start sample similar to generic package
```

Comment:
duplicated code, also it's worth documenting the problem that it solves, why we launch it from the command line

Reply:
check class documentation.
I still want to know that I'm originating from specific package

Reply:
> I still want to know that I'm originating from specific package

extract the logic and use it in the appropriate packages then? duplicated code should be avoided